11-type-constructors/algo/test/Tests.hs

```diff
@@ -0,0 +1,252 @@
+{-# LANGUAGE PartialTypeSignatures #-}
+
+import Algo hiding (syn, tequiv)
+import qualified Algo (syn, tequiv)
+import Data.Maybe
+import Parser
+import Test.HUnit
+
+main :: IO ()
+main = runTestTTAndExit tests
+
+tests :: Test
+tests =
+  TestList . concat $
+  [tequivTests, synTests, freshTests, fresh2Tests, αKndTests]
+  where
+    tequivTests =
+      [ Nil ⊢ tequiv Bse Bse Type ~?= True
+      , Nil ⊢ tequiv Bse Bse (S Type Bse) ~?= True
+      , Nil ⊢ tequiv Bse Bse (Π "t" Type (S Type (TVar "t"))) ~?= False
+      , Nil ⊢ tequiv (Bse :⊕ Bse) (Bse :⊕ Bse) Type ~?= True
+      , Nil ⊢ tequiv (Bse :⊕ Bse) (Bse :⊕ Bse) (S Type $ Bse :⊕ Bse) ~?= True
+      , Nil ⊢
+        tequiv (Bse :⊕ Bse) (Bse :⊕ Bse) (S (S Type $ Bse :⊕ Bse) (Bse :⊕ Bse)) ~?=
+        True
+      , Nil ⌢ ("Int", Type) ⊢ tequiv (TVar "Int") Bse Type ~?= False
+      , Nil ⌢ ("Int", Type) ⊢ tequiv (TVar "Int") (TVar "Int") Type ~?= True
+      , Nil ⌢ ("Int", Type) ⊢
+        tequiv
+          ((TVar "Int") :⊕ (TVar "Int"))
+          ((TVar "Int") :⊕ (TVar "Int"))
+          Type ~?=
+        True
+      , Nil ⌢ ("Int", Type) ⌢ ("Int'", S Type $ TVar "Int") ⊢
+        tequiv
+          ((TVar "Int'") :⊕ (TVar "Int"))
+          ((TVar "Int") :⊕ (TVar "Int"))
+          Type ~?=
+        True
+      , Nil ⌢ ("Int", Type) ⌢ ("Int'", S Type $ TVar "Int") ⊢
+        tequiv
+          ((TVar "Int'") :⊕ (TVar "Int'"))
+          ((TVar "Int") :⊕ (TVar "Int"))
+          Type ~?=
+        True
+      , Nil ⌢ ("Int", Type) ⌢ ("Int'", S Type $ TVar "Int") ⊢
+        tequiv
+          ((TVar "Int") :⊕ (TVar "Int'"))
+          ((TVar "Int'") :⊕ (TVar "Int"))
+          Type ~?=
+        True
+      , Nil ⌢ ("Int", Type) ⌢ ("Int'", S Type $ TVar "Int") ⊢
+        tequiv
+          ((TVar "Int") :⊕ (TVar "Int'"))
+          ((TVar "Int'") :⊕ (TVar "Int'"))
+          Type ~?=
+        True
+      , Nil ⌢ ("Int", Type) ⌢ ("Int'", S Type $ TVar "Int") ⊢
+        tequiv
+          ((TVar "Int") :⊕ (TVar "Int'"))
+          ((TVar "Int'") :⊕ (TVar "Bse"))
+          Type ~?=
+        False
+      , Nil ⊢ tequiv (Tλ "t" Type $ TVar "t") (Tλ "t" Type Bse) Type ~?= False
+      , Nil ⊢
+        tequiv (Tλ "t" Type $ TVar "t") (Tλ "t" Type Bse) (Π "t" Type Type) ~?=
```

Comment:
"t" isn't in scope in the annotation on Tλ right?

Reply:
Yes, although I'm confused what that has to do with those tests?